### PR TITLE
TDSW-21: creating a grey section

### PR DESCRIPTION
**Pull request by jenbreese**
# READY FOR REVIEW

# Summary
- Adding a section variant that adds a grey background or a "card" background

# Review By (Date)
- 6/12

# Criticality
- Normal

# Urgency
- Normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Create a basic page.
3. Add a section and try out the grey and card options.
4. The optional padding has been removed by plan per design.
5. Screenshot 2024-06-06 at 4 25 41 PM

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
[- TDSW-21](https://stanfordits.atlassian.net/browse/TDSW-21)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: sccr_subtheme.react_behaviors.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - paragraphs_type|stanford_card
+  config:
+    card_border_style:
+      type: select
+      title: 'Card Border Style'
+      description: Select the border style for displaying this card.
+#      default_value: su-card-wrapper--border-regular
+      options:
+        su-card-wrapper--border-none: Minimal (no border)
+        su-card-wrapper--border-regular: Border and Drop Shadow
```
Comment (jdwjdwjdw) [inline sccr_subtheme.react_behaviors.yml]: Do want this card variant code included in this PR?

File: src/scss/layout/index.scss [hunk 2]
```diff
@@ -19,4 +19,5 @@
 //
 
 // Config and supporting scss.
-// @import 'directory/file';
+// @import 'full-width';
```
Comment (jdwjdwjdw) [inline src/scss/layout/index.scss]: Is this needed?

File: src/scss/theme/_section-backgrounds.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// .sccr-section-pb-none
+
+.sccr-section-color--light-grey {
+    @include responsive-spacing('padding', 2);
+
+    background-color: $su-color-foggy-light;
+}
+
+.sccr-section-color--card {
+    @include flex-column(10 of 12);
```
Comment (jdwjdwjdw) [inline src/scss/theme/_section-backgrounds.scss]: I'm curious why you added this flex-column mixin? It's applying a max-width which makes the card smaller than the grey version, and stops the card from being aligned with the page title max-width, which is how it looks in the Figma image.
Screenshot 2024-06-11 at 2 48 00 AM

File: sccr_subtheme.react_behaviors.yml (hunk 1 above)
Comment (jenbreese) [reply sccr_subtheme.react_behaviors.yml]: No it just was part of the starter.

File: src/scss/layout/index.scss (hunk 2 above)
Comment (jenbreese) [reply src/scss/layout/index.scss]: Not really. It is just an example.

File: src/scss/theme/_section-backgrounds.scss (hunk 3 above)
Comment (jenbreese) [reply src/scss/theme/_section-backgrounds.scss]: Um, I no longer remember so I removed it.

**Review by jdwjdwjdw — APPROVED** [review]
Looks good, thanks!

**Pull request merged**